src/libNode/Node.cpp

```diff
@@ -706,15 +706,29 @@ bool Node::ProcessTxnPacketFromLookup(
 
   uint64_t epochNumber = 0;
   uint32_t shardId = 0;
+  PubKey lookupPubKey;
   vector<Transaction> transactions;
 
   if (!Messenger::GetNodeForwardTxnBlock(message, offset, epochNumber, shardId,
-                                         transactions)) {
-    LOG_EPOCH(WARNING, to_string(m_mediator.m_currentEpochNum).c_str(),
+                                         lookupPubKey, transactions)) {
+    LOG_EPOCH(WARNING, std::to_string(m_mediator.m_currentEpochNum).c_str(),
               "Messenger::GetNodeForwardTxnBlock failed.");
     return false;
   }
 
+  auto lookupNodes = m_mediator.m_lookup->GetLookupNodes();
```

Comment:
`lookupNodes` is a reference so the operation is now on a shared resource if the function `ProcessTxnPacketFromLookup` is invoked from multiple threads. Can you check if any race condition could happen here?

Reply:
Ok, i will put it to a const reference, there will be only query from it.